bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -62,63 +57,83 @@ public void unsetScriptExtensionManager(ScriptExtensionManager scriptExtensionMa
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    public void addScriptEngineFactory(ScriptEngineFactory provider) {
-        this.scriptEngineFactories.add(provider);
+    public void addScriptEngineFactory(ScriptEngineFactory engineFactory) {
+        List<String> scriptTypes = engineFactory.getScriptTypes();
 
-        for (String language : provider.getLanguages()) {
-            this.supportedLanguages.put(language, provider);
+        logger.trace("{}.getScriptTypes(): {}", engineFactory.getClass().getSimpleName(), scriptTypes);
+        for (String scriptType : scriptTypes) {
+            if (isCustomFactory(engineFactory)) {
+                this.customSupport.put(scriptType, engineFactory);
+            } else {
+                this.genericSupport.put(scriptType, engineFactory);
+            }
+        }
+        logger.info("Added {}", engineFactory.getClass().getSimpleName());
+        for (javax.script.ScriptEngineFactory f : ScriptEngineFactory.engineManager.getEngineFactories()) {
+            logger.debug(
+                    "ScriptEngineFactory details for {} ({}): supports {} ({}) with file extensions {}, names {}, and mimetypes {}",
+                    f.getEngineName(), f.getEngineVersion(), f.getLanguageName(), f.getLanguageVersion(),
+                    f.getExtensions(), f.getNames(), f.getMimeTypes());
         }
     }
 
-    public void removeScriptEngineFactory(ScriptEngineFactory provider) {
-        this.scriptEngineFactories.remove(provider);
+    public void removeScriptEngineFactory(ScriptEngineFactory engineFactory) {
+        List<String> scriptTypes = engineFactory.getScriptTypes();
 
-        for (String language : provider.getLanguages()) {
-            this.supportedLanguages.remove(language, provider);
+        logger.trace("{}.getScriptTypes(): {}", engineFactory.getClass().getSimpleName(), scriptTypes);
+        for (String scriptType : scriptTypes) {
+            if (isCustomFactory(engineFactory)) {
+                this.customSupport.remove(scriptType, engineFactory);
+            } else {
+                this.genericSupport.remove(scriptType, engineFactory);
+            }
         }
+        logger.info("Removed {}", engineFactory.getClass().getSimpleName());
```

Comment:
Reduce this from `info` to `debug` or `trace`.

Reply:
Done